### ISSUE TITLE
Bring your own identity management provider with App Center Auth

### ISSUE THREAD

**Issue by amchew** [microsoft/appcenter#637]
App Center Auth in early preview enables developers to connect to their Azure AD B2C tenant. However, we fully expect our developers to use other popular identity and access management providers such as Auth0, Firebase Authentication, and Ping Identity. To broaden our reach and support for Auth and Data, we will add the feature for developers to bring their own identity management providers with support for our MBaaS services. Developers will be able to connect their own identity management providers and use the JSON Web Token returned to identify the signed in user (via `account_id`). 

Leave a comment below on the identity access and management providers you are using, and any additional experiences that you may need!

Here are some sample designs: 

![image](https://user-images.githubusercontent.com/33160470/62172004-c68e9a00-b2e5-11e9-8ce1-bad0add519ba.png)

For Auth0: 
![image](https://user-images.githubusercontent.com/33160470/62172094-02296400-b2e6-11e9-8559-aaca106f7443.png)

For Firebase: 
![image](https://user-images.githubusercontent.com/33160470/62172044-e02fe180-b2e5-11e9-89c0-9d20907f4243.png)

Landing page:
![image](https://user-images.githubusercontent.com/33160470/62172062-ec1ba380-b2e5-11e9-877f-bb0d6508e13b.png)

**Comment by Behrad-iz** [comment]
Do you guys have an estimation when this feature will be available?
We need to use Auth0
Thanks

**Comment by amchew** [comment]
Hey @Behrad-iz! We have the work queued up for Bring your own Identity with Auth0 in the next couple of months, and is next on the priority list. I'd love to hear more about your use case, would you be keen on chatting some time this or next week? Email me at amchew [at] microsoft [dot] com if so!

**Comment by amchew** [comment]
Hi @behrad-iz, thanks again for chatting today! The work for Bring your own Identity with Auth0 and Firebase Authentication has just been picked up by our engineering team today, and will keep you updated on the progress! For the rest of the folks, we're looking for beta testers to give feedback, so feel free to comment here if you'd like to help us shape our product.

**Comment by amchew** [comment]
Thanks a lot @galfonso777 for the feedback! Will connect with you via email.

**Comment by StarbuckSapien** [comment]
The company I'm working with is using Amazon Cognito User Pools for B2C user authentication for their new project.

**Comment by amchew** [comment]
Hi @StarbuckSapien, thanks for commenting :) Would you be free next week for a call for me to understand more about your scenarios? My email is amchew [at] microsoft [dot] com, email me if you're keen!

**Comment by amchew** [comment]
Hey @gaurap, thanks for reaching out! The experience is easy when you connect your Firebase project to Auth, you just have to input the project id and we'll collect all the information from Firebase. You won't have to input `client_id` or `client_secret`.

Here's a sneak peek of how it looks like: 
![image](https://user-images.githubusercontent.com/33160470/65805018-ef41dd00-e138-11e9-86f1-4d0f5f5ef0b5.png)
![image](https://user-images.githubusercontent.com/33160470/65805051-04b70700-e139-11e9-8229-ac96e503eeac.png)
![image](https://user-images.githubusercontent.com/33160470/65805096-2912e380-e139-11e9-8a9d-c8a532d60786.png)

Once the user has signed in to the app, and the identity is stored in the Firebase tenant, and you'll get back an account_id in App Center that you can uniquely identify that user.

**Comment by akamud** [comment]
Our client has dropped the idea of using AppCenter Data because he has his own IdentityServer implemented and there is no way to use it as a identity provider for AppCenter Auth right now. At least not without having to maintain to users database (his own and Azure B2C)

Any plans on supporting any OpenID provider? What would be the difference for Auth since he exposes all the .well-known urls?

**Comment by akamud** [comment]
Our client has dropped the idea of using AppCenter Data because he has his own IdentityServer implemented and there is no way to use it as a identity provider for AppCenter Auth right now. At least not without having to maintain to users database (his own and Azure B2C)

Any plans on supporting any OpenID provider? What would be the difference for Auth since he exposes all the .well-known urls?

**Comment by akamud** [comment]
> Any plans on supporting any OpenID provider? What would be the difference for Auth since my client’s IS exposes all the .well-known urls?

Can anyone comment on this questions please?

**Comment by bnoffer** [comment]
I can second @akamud 's request, because our company has developed its own IDP. We could build a solution as a Federated IDP with Azure AD B2C, but I think eliminating the additional layer would be preferred so we do not generate two different user databases.

**Comment by akamud** [comment]
> I can second @akamud 's request, because our company has developed its own IDP. We could build a solution as a Federated IDP with Azure AD B2C, but I think eliminating the additional layer would be preferred so we do not generate two different user databases.

That is exactly what we are trying to avoid.

**Comment by akamud** [comment]
Hi @amchew, can you please provide any information about our requests?

**Comment by marthinfreij** [comment]
Would be nice with support for [Sign in with Apple](https://developer.apple.com/sign-in-with-apple/).

**Comment by chris-ux** [comment]
Any ETA on Auth0 Integration at least?

**Comment by galfonso777** [comment]
@amchew Is App Center still being iterated and/or supported? Haven't seen any updates to iteration plans since December and haven't see any roadmap for 2020. This is one of the several features that my company is monitoring which will determine how implement Auth moving forward with all our applications. Any update you could provide would be much appreciated. Thank you.

**Comment by Zakeelm** [comment]
Thank you so much for your feedback. Unfortunately, this work won't be completed. We announced this week in a [blog post]( https://devblogs.microsoft.com/appcenter/app-center-mbaas-retirement/) that we’re retiring the Auth and Data services later this year and recommending customers transition to Azure Active Directory B2C and Azure Cosmos DB instead. Let me know if you have any additional questions about this transition and i'll make sure to get them answered! Closing this issue.
